### ISSUE TITLE
E3SM doesn't allow specifying the run stop date

### ISSUE THREAD

**Issue by tangq** [E3SM-Project/E3SM#4647]
E3SM doesn't automatically adjust the last submission length to stop at the desired date.

A `STOP_UNTIL_DATE` option would be helpful to have, so before the model automatically resubmit the last job, it will modify the `STOP_N` and `RESUBMIT` options to stop at a specific date.

**Comment by rljacob** [comment]
Are you using STOP_OPTION of "date" and STOP_DATE?

**Comment by tangq** [comment]
No, I use years for `STOP_OPTION`. But when running the 165 years historical run for DECK, `STOP_N=20` is optimal for the throughput, but it won't stop at 2015/01/01. Instead, it ends up stopping at 2015/12/14, when the input file for historical runs is not available and the run fails. The extra year 2015 simulation waste the resource and causes issues for post-processing.

We need a way to specify the date when the run should stop. CIME or some other mechanism should be able to shorten the length of the last submission based on the start date or the restart date and the stop date.

**Comment by jedwards4b** [comment]
You would need to change the STOP_OPTION from years to date for the last leg of the run.

**Comment by tangq** [comment]
@jedwards4b , thanks for the suggestion. We certainly can do it manually for the last submission, but it would be nicer if it is automatic. The model knows the date it stops at for the previous submission and with a `STOP_UNTIL_DATE` option, it knows all the information to automatically adjust `STOP_OPTION` and `STOP_N` before submitting the last job.

Just a good feature to have for E3SM. According to @golaz , this feature is available for many other models.

**Comment by golaz** [comment]
I support @tangq 's suggestion. It would be really nice (and it would make our lives as model runners easier) if CIME was just a little smarter: we tell it to run the model for 165 years, in segments of 20 years or less. CIME figures out how many re-submissions are needed, and the length of the last segment.

**Comment by rljacob** [comment]
So instead of specifying RESUBMIT, a user would specify total years (165) and a segment length (20) and CIME would calculate RESUBMIT  and make sure the last submission didn't overrun the total.

**Comment by rljacob** [comment]
A CIME issue was opened for this https://github.com/ESMCI/cime/issues/4126

@tangq Erik suggested that just defining STOP_YEAR might work.  You would do:
./xmlchange STOP_YEAR=20150101
and rerun your last 20 year segment.

**Comment by tangq** [comment]
@rljacob , we can try ./xmlchange STOP_YEAR=20150101 for the next RRM historical run to check if it still works.

Tagging @golaz and @singhbalwinder .

**Comment by rljacob** [comment]
Yes you're right its STOP_DATE.
